darts/utils/likelihood_models/sklearn.py

```diff
@@ -412,59 +414,279 @@ def _get_median_prediction(self, model_output: np.ndarray) -> np.ndarray:
+        for i, params_proba in enumerate(model_output):
+            component_index = i % n_component
+            output_index = i // n_component
+            class_proba[
+                output_index,
+                :,
+                self._index_first_param_per_component[
+                    component_index
+                ] : self._index_first_param_per_component[component_index + 1],
+            ] = params_proba
+
+        # shape (output_chunk_length, n_series * n_samples, n_likelihood_parameters)
+        # n_likelihood_parameters is the sum of the number of classes for each component
+        # n_likelihood_parameters are ordered as in component_names:
+        # [comp0_l1, comp0_l2, comp1_l1, comp1_l2, comp1_l3,...]
+        # should be accessed through _index_first_param_per_component (for comp1 => 2)
+        return class_proba
+
+    def sample(self, model_output: np.ndarray) -> np.ndarray:
+        """
+        Samples a prediction from the likelihood distribution and the predicted parameters.
+        """
+        # shape (output_chunk_length, n_series * n_samples, n_likelihood_parameters)
+        n_output, n_samples, _ = model_output.shape
+
+        n_component = len(self._classes) // self._n_outputs
+
+        # shape (output_chunk_length, n_series * n_samples, n_components)
+        preds = np.empty((n_output, n_samples, n_component), dtype=int)
+
+        # Some models have an approximation error, the probabilities are adjusted
+        # if their total is below the 1e-7 tolerance threshold around 1.
+        for component_idx, (component_start, component_end) in enumerate(
+            zip(
+                self._index_first_param_per_component[:-1],
+                self._index_first_param_per_component[1:],
+            )
+        ):
+            component_probabilities = model_output[:, :, component_start:component_end]
+            total_proba = component_probabilities.sum(
+                axis=2
+            )  # shape  (n_output, n_samples)
+            difference = np.ones((n_output, n_samples)) - total_proba
+            tolerance = 1e-7
+            if np.any(np.abs(difference) > tolerance):
+                raise_log(
+                    ValueError(
+                        "The class probabilities returned by the model do not sum to one"
+                    )
+                )
+
+            component_probabilities += (
+                difference[:, :, np.newaxis] / component_probabilities.shape[2]
+            )
+
+            # argmax stops at the first True
+            # first time the random sample is greater than cumulative probability
+            # [0.2, 0.1, 0.5, 0.2] -> [0.2, 0.3, 0.8, 1]
+            # random: 0.7 -> idx = 2, this outcomes has 0.5 probability of happening
+            sampled_idx = np.argmax(
+                self._rng.uniform(0, 1, size=(difference.shape))[:, :, np.newaxis]
+                < np.cumsum(component_probabilities, axis=2),
```

Comment:
I guess `<=` would include everything (also `p==1.0`)
```suggestion
                <= np.cumsum(component_probabilities, axis=2),
```